tdo.sh

```diff
@@ -43,6 +43,43 @@ For more information, visit https://github.com/2kabhishek/tdo
 EOF
 }
 
+config_setup() {
+    # List of variable names to check
+    local env_vars=("TIMESTAMP_ENTRY" "TIMESTAMP_NEWNOTE" "ENTRY_TIMESTAMP" "NOTE_TIMESTAMP" "FILE_NAME_AS_TITLE")
+
+    # Check if any of the variables are set as environment variables
+    local env_variables_set=false
+    for var in "${env_vars[@]}"; do
+        if [ ! -z "${!var}" ]; then
+            env_variables_set=true
+            break
+        fi
+    done
+
+    # Function to source the configuration file if it exists
+    source_config_file() {
+        local config_file="$HOME/.config/tdorc"
+        if [ -f "$config_file" ]; then
+            source "$config_file"
+        fi
+    }
+
+    # If none of the variables are set as environment variables, source the config file
+    if [ "$env_variables_set" = false ]; then
+        source_config_file
+    fi
+}
+
+# validate whether a variable is set to true or false, and set to default otherwise
+validate_and_set() {
+    local default_val="${2:-true}"
+    local var_value="${1:-$default_val}"
+    if [[ "$var_value" != "true" && "$var_value" != "false" ]]; then
+        var_value=$default_val
+    fi
+    echo "$var_value"
+}
+
```

Comment:
I think the config loading can be improved a bit, rather than doing a either/or between tdorc and env, we can actually do both

here's some suggestions:

- source tdorc at the beginning of executing, this will add all the configs present there to env, the missing ones will be excluded
- store the config options as global level variables and get their values from env / assing a default value, for example:
```bash
add_timestamp_entry="${ADD_TIMESTAMP_ENTRY:-$true}"
```

- and then use these variables in the functions throughout the script

I know this will override any existing env configs if they are also present on tdorc but I am ok with that trade off

Reply:
Done! I like this approach better actually as it simplifies the logic in config_setup quite a bit. Thanks for the suggestion!